portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,39 +14,71 @@
 
 package com.google.sps.servlets;
 
-import com.google.gson.Gson;
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
-import java.util.*;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.*;
 
 /** Servlet that encapsulates comments. */
 @WebServlet("/data")
 public final class DataServlet extends HttpServlet {
   @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      response.setContentType("application/json");
-      String json = new Gson().toJson(comments);
-      response.getWriter().println(json);
+      Query query = new Query("Comment");
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      PreparedQuery results = datastore.prepare(query);
+
+      List<Comment> comments = new ArrayList<>();
+      for (Entity entity : results.asIterable()) {
+        long id = entity.getKey().getId();
+        String text = (String) entity.getProperty("text");
+        String name = (String) entity.getProperty("name");
+        Comment comment = new Comment(id, text, name);
+        comments.add(comment);
+      }
+
+      Gson gson = new Gson();
+
+      response.setContentType("application/json;");
+      response.getWriter().println(gson.toJson(comments));
     }
 
   @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
       // Get the input from the form.
       String userComment = request.getParameter("user-comment");
+      String userName = request.getParameter("user-name");
 
       Entity commentEntity = new Entity("Comment");
-      commentEntity.setProperty("comment", userComment);
+      commentEntity.setProperty("text", userComment);
+      commentEntity.setProperty("name", userName);
 
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
       datastore.put(commentEntity);
 
       // Redirect back to the HTML page.
       response.sendRedirect("/comment.html");
     }
-}
+
+    public class Comment {
```

Comment:
Right now, `Comment` is a nested inner class of `DataServlet`.

> Non-static nested classes (inner classes) have access to other members of the enclosing class, even if they are declared private.  (https://docs.oracle.com/javase/tutorial/java/javaOO/nested.html)

I recommend to either A) create a separate file defining the Comment class (and encapsulate the fields with `private`) which avoids such contemplation altogether─or B) alternatively make the nested class private and static here to clarify where it's intended to be used.